adapter_files/precice.cpp

```diff
@@ -259,10 +264,13 @@ double Precice::initialize() {
       solverInterface.setMeshVertices(meshID[indexMarkerWetMappingLocalToGlobal[i]], vertexSize[i], coords,
                                       vertexIDs[i]);
       forceID[indexMarkerWetMappingLocalToGlobal[i]] = solverInterface.getDataID(
-          "Forces" + to_string(indexMarkerWetMappingLocalToGlobal[i]), meshID[indexMarkerWetMappingLocalToGlobal[i]]);
-      displDeltaID[indexMarkerWetMappingLocalToGlobal[i]] =
-          solverInterface.getDataID("DisplacementDeltas" + to_string(indexMarkerWetMappingLocalToGlobal[i]),
-                                    meshID[indexMarkerWetMappingLocalToGlobal[i]]);
+          preciceWriteDataName +
+              (indexMarkerWetMappingLocalToGlobal[i] == 0 ? "" : to_string(indexMarkerWetMappingLocalToGlobal[i])),
+          meshID[indexMarkerWetMappingLocalToGlobal[i]]);
+      displDeltaID[indexMarkerWetMappingLocalToGlobal[i]] = solverInterface.getDataID(
+          preciceReadDataName +
+              (indexMarkerWetMappingLocalToGlobal[i] == 0 ? "" : to_string(indexMarkerWetMappingLocalToGlobal[i])),
+          meshID[indexMarkerWetMappingLocalToGlobal[i]]);
```

Comment:
I'm not sure if omitting the zero here is right due to the local to global index mapping here. An index could be zero on a local mesh although we have multiple interfaces. As an alternative, I could also ask `i` to be zero instead of `indexMarkerWetMappingLocalToGlobal[i]`. Any suggestions?

Reply:
I am not sure I understand what is going on here and what you mean.

Could you give an example of where this could fail?

Reply:
I'm currently trying to get a clear decision here and run the case in parallel. However, the case doesn't work in parallel. I tried with the current `develop` state and it doesn't work as well (MPI error). May I ask you to run one of our current tutorials in parallel using SU2 in order to confirm?

Reply:
I can run (at least for a few runtime seconds) the tutorials from develop with SU2 in parallel (`runFluid -parallel`) with preCICE v2.2.0 on Ubuntu 20.10.

Reply:
It was an mpi failure of my installation..

I tested now the new configuration in serial and parallel using a single interface and it works. I think I also understand now what the original author of this code has done here and why he used `indexMarkerWetMappingLocalToGlobal[i]`. Then, the current implementation is the right one. In order to fully test it, one would need to set up a case with more than one interface and run it in parallel.

Reply:
Any objections before merging @MakisH ? 
There is an exception left when the simulation has finished, I will merge it here.

Reply:
Wait, I will not fix it here, since there are actually different issues on this topic and they seem to be a bigger problem.

Reply:
An explanation / answer to my original question is still missing here:
> I am not sure I understand what is going on here and what you mean.
Could you give an example of where this could fail?

Other than that, feel free to merge.

Reply:
An explanation / answer to my original question is still missing here:
> I am not sure I understand what is going on here and what you mean.
Could you give an example of where this could fail?

Other than that, feel free to merge.

Reply:
> Could you give an example of where this could fail?

The example was meant to be this sentence

> In order to fully test it, one would need to set up a case with more than one interface and run it in parallel.

So, to explain it more in detail: the `indexMarkerWetMappingLocalToGlobal` stores the global appended index belonging to the (sequential) number of the locally owned interface number.  Not every process works on every global interface and the adapter loops in the end only over locally owned interfaces. Hope this makes it more clearly.

Reply:
Yes, this makes sense, thanks!